checkQC/handlers/q30_handler.py

```diff
@@ -28,12 +28,18 @@ def collect(self, signal):
 
     def check_qc(self):
 
+        index_count = 0
+        non_index_count = 0
         for error_dict in self.error_results:
             lane_nbr = int(error_dict["lane"])
-            read = error_dict["read"]
             percent_q30 = error_dict["percent_q30"]
-            is_index_read = error_dict["percent_q30"]
+            is_index_read = error_dict.get("is_index_read", False)
             read_or_index_text = "index read" if is_index_read else "read"
+            # Differentiate read values for indexed from non-indexed reads
+            index_count += 1 if is_index_read else index_count
```

Comment:
I think you need to change the last `index_count` to 0. We want the counter to NOT be increased if we loop over a non index read, not to add the current `index_count`. I think this would have yield an inaccurate count if the `TestQ30Handler.setUp` looked like this: 

```
        value_1 = {"lane": 1, "read": 1, "percent_q30": 82}
        value_2 = {"lane": 1, "read": 2, "percent_q30": 50, "is_index_read": True}
        value_3 = {"lane": 1, "read": 3, "percent_q30": 82}
        value_4 = {"lane": 1, "read": 4, "percent_q30": 50, "is_index_read": True}
```